drain/drain.py

```diff
@@ -0,0 +1,417 @@
+            continue
+
+        plugin.log("Sending over %d hops to %s %s using %s fees" % (len(route), payload['command'], payload['amount'], fees))
+        for r in route:
+            plugin.log("    - %s  %14s  %s" % (r['id'], r['channel'], r['amount_msat']))
+
+        try:
+            plugin.rpc.sendpay(route, payment_hash, label)  # , payload['amount'] - fees)
+            result = plugin.rpc.waitsendpay(payment_hash, payload['retry_for'] + start_ts - int(time.time()))
+            if result.get('status') == 'complete':
+                payload['success_msg'] += ["%dmsat sent over %d hops to %s %dmsat [%d/%d]" % (payload['amount'] + fees, len(route), payload['command'], payload['amount'], chunk+1, payload['chunks'])]
+                # we need to wait for gossipd to update to new state,
+                # so remaining amounts will be calculated correctly for the next chunk
+                spendable, _ = spendable_from_scid(plugin, payload, payload['scid'])
+                while spendable == spendable_before:
+                    time.sleep(0.5)
+                    spendable, _ = spendable_from_scid(plugin, payload, payload['scid'])
+                return True
+            return False
+
+        except RpcError as e:
+            erring_message = e.error.get('message', '')
+            erring_channel = e.error.get('data', {}).get('erring_channel')
+            erring_index = e.error.get('data', {}).get('erring_index')
+            erring_direction = e.error.get('data', {}).get('erring_direction')
+
+            # detect exceeding of HTLC commitment fee
+            if 'Capacity exceeded' in erring_message and erring_index == 0:
+                match = HTLC_FEE_PAT.search(erring_message);
+                if match:  # new servers tell htlc_fee via exception (#2691)
+                    raise ValueError("htlc_fee is %s" % match.group(1))
+                raise ValueError("htlc_fee unknown")
+
+            if erring_channel == payload['scid']:
+                raise RpcError(payload['command'], payload, {'message': 'Error with selected channel: %s' % erring_message})
+
+            plugin.log("RpcError: " + str(e))
+            if erring_channel is not None and erring_direction is not None:
+                excludes.append(erring_channel + '/' + str(erring_direction))
+
+
+def execute(command: str, scid: str, percentage: int,
+        chunks: int, maxfeepercent: float, retry_for: int, exemptfee: Millisatoshi):
+
+    payload = {
+        "command" : command,
+        "scid": scid,
+        "percentage": percentage,
+        "chunks": chunks,
+        "maxfeepercent": maxfeepercent,
+        "retry_for": retry_for,
+        "exemptfee": exemptfee,
+
+        "labels" : [],
+        "success_msg" : [],
+    }
+
+    # check parameters
+    if command != 'drain' and command != 'fill':
+        raise RpcError(payload['command'], payload, {'message': 'Invalid command. Must be drain or fill.'})
+    if percentage < 0 or percentage > 100:
```

Comment:
We can change back, once the plugin converts a "fill 0%" to a "drain 100%". Currently it will trow an error as the target is above/below spendable/receivable.